src/Bearded.TD/Game/Simulation/Elements/ElementalPhenomenonScopeBase.cs

```diff
@@ -92,36 +87,44 @@ private void startEffect(TEffect effect, out ElementalStatus? newStatus)
 
     private void endEffectIfPreviouslyActive()
     {
-        if (activeEffect is null) return;
+        if (activeEffect is not { } current) return;
 
-        EndScope(target);
-        activeEffect.StatusIcon?.DeleteImmediately();
+        EndScope(current.Effect);
+        current.StatusIcon?.DeleteImmediately();
         activeEffect = null;
     }
 
     private void updateStatusIconExpiry(IStatusReceipt statusIcon)
     {
-        var latestExpiry = activeEffects.Select(e => e.Expiry).Max();
+        var latestExpiry = activeEffects.Max(static e => e.Expiry);
         statusIcon.SetExpiryTime(latestExpiry);
     }
 
-    protected abstract bool TryChooseEffect(out TEffect effect);
-    protected abstract void StartScope(GameObject target, out ElementalStatus? status);
-    protected abstract void StartEffect(GameObject target, TEffect effect, EffectStartContext context);
-    protected abstract void ApplyEffectTick(GameObject target, TEffect effect);
-    protected abstract void EndEffect(GameObject target, TEffect effect);
-    protected abstract void EndScope(GameObject target);
+    protected abstract TEffect? TryChooseEffect(ReadOnlySpan<EffectWithExpiry> activeEffects);
```

Comment:
nit: the `Try` method pattern is to return a boolean and use an `out` variable. Since this was changed to a direct access, should we rename this to `ChooseEffect`?